core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/named/cache/NamedOperationCache.java

```diff
@@ -229,6 +230,7 @@ public NamedOperationDetail getFromCache(final String name) throws CacheOperatio
         throw new CacheOperationFailedException(String.format("No named operation with the name %s exists in the cache", name));
     }
 
+    @SuppressFBWarnings(value = "DCN_NULLPOINTER_EXCEPTION", justification = "Investigate an improved null checking approach")
```

Comment:
Could this just be an Objects.isNull check/throw like for name below it?

Reply:
I didn't look into this one very closely, but I think the `NullPointerException` originates in the `getOperationName()` method's class and not this class. Looking at that class (`NamedOperationDetail.java`), it seems there's a situation where the `String` field returned is unset - which is not the same as having a value of null, causing a null pointer. This explains why there's that separate `if` check with the `isNull` for way the value is set but is null.

This cannot be fixed in this class I don't think. This is the only place where `getOperationName()` is used, so this method could (and probably should) be changed to not have this problem.

Reply:
It requires more investigation, but if the `NamedOperationDetail` is itself null, that could be causing this, in which case adding an initial check for that would fix this problem.